scripts/beautify.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+# Reformat C++ code using clang-format
+
+# note: with the -style=file option, clang-format reads the config from ./.clang-format
+# See here for the full list of supported options: https://clang.llvm.org/docs/ClangFormatStyleOptions.html
+FORMAT_CMD="clang-format -i -style=file"
+
+# Filter out any files that shouldn't be auto-formatted.
+# note: -v flag inverts selection - this tells grep to *filter out* anything
+#       that matches the pattern. For testing, you can remove the -v to ssee
+#       which files would have been excluded.
+FILTER_CMD="grep -v -E objects|src/polyclipping|src/CGAL_Nef3_workaround.h|src/CGAL_workaround_Mark_bounded_volumes.h|src/convex_hull_3_bugfix.h|src/OGL_helper.h"
+
+# C++ file extensions
+SRC_PATTERN=".*\.(h|hpp|cc|cpp)"
+
+function reformat_all() {
+    find . -regextype posix-extended -regex "$SRC_PATTERN" \
```

Comment:
`-regextype` is a GNU extension and won't work on BSD-based systems like macOS.

Reply:
Darn, that's unfortunate... Now that you mention it, I think I've hit this issue before on my mac

Reply:
`find -E` appears to have the equivalent function on BSD.

Reply:
I changed it back to the way I had it before.

Reply:
Unfortunately, it doesn't look like `-E` works on gnu `find` :/